### PR TITLE
Issue 106

### PR DESCRIPTION
**Pull request by pralhad88**
POST /on_assessment/details/{enrolmentKey}  it is update students data in once.

### CODE REVIEW

File: lib/routes/onAssessment.js [hunk 1]
```diff
@@ -82,19 +82,20 @@ module.exports = [
...
-
-
-                return {'sucess': true}
+                // once student completed his test then no one given the test using same enrolment key.
+                if (enrolmentKey.student.stage != 'completedTest') {
+                    let payload = studentService.swapEnumKeysWithValues(request.payload);
+                    await assessmentService.patchStudentDetails(enrolmentKey, payload);
+                    return {'sucess': true}
+                }else{
+                    return {'Error': "This enrolment key is already used by another student. Please call to Navgurukuls  helpline number and take new test." }
```
Comment (RishabhVerma) [inline lib/routes/onAssessment.js]: Please read about Boom and see how errors are thrown in our code base. Best if we maintain that standard.

**Comment by RishabhVerma** [comment]
Please check the comments and fix them.

**Comment by pralhad88** [comment]
it is done throwing errors using Boom.

**Comment by RishabhVerma** [comment]
Pralhad the current way to fix this bug wouldn't be enough. Because currently our flow looks like this:

1. Student enters details (DOB, name etc.)
2. Student answers test
3. Student enters more details (city, state etc.)

Now the issue is that his stage is marked as `completedTest` after step 2. And with the logic you've implemented the person won't be able to add the details in step 3. So basically you will have to do something like this:

1. After step 1 change stage of student to `basicDetailsEntered`
2. After step 2 change the stage of student to `completedTest`
3. At step 3 change the stage to `completedTestWithDetails`

**Pull request merged**